emukit/core/loop/candidate_point_calculators.py

```diff
@@ -69,24 +73,33 @@ def _get_information_source_parameter(self) -> InformationSourceParameter:
             raise ValueError('No source parameter found')
         return source_parameter[0]
 
-    def compute_next_points(self, loop_state: LoopState=None) -> np.ndarray:
+    def compute_next_points(self, loop_state: LoopState=None, context: dict=None) -> np.ndarray:
         """
         Computes the location and source of the next point to evaluate by finding the optimum input location at each
         information source, then picking the information source where the value of the acquisition at the optimum input
         location is highest.
 
         :param loop_state: Object that tracks the state of the loop. Currently unused
+        :param context: Contains variables to fix through optimization of acquisition function. The dictionary key is
+                        the parameter name and the value is the value to fix the parameter to.
         :return: A list of function inputs to evaluate next
         """
         f_mins = np.zeros((len(self.source_parameter.domain)))
         x_opts = []
 
+        if context is None:
+            context = dict()
```

Comment:
Can we just make this the default value in the method signature?

Reply:
I thought having mutable default arguments was dangerous somehow:

https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Reply:
Wow, I learned something new today. Thanks, this answers